src/main/java/datastructure/NodeGraph.java

```diff
@@ -0,0 +1,88 @@
+package datastructure;
+
+import java.util.ArrayList;
+
+/**
+ * Created by 101010.
+ */
+public class NodeGraph {
```

Comment:
Maybe change the name to just Graph? Since graph naturally consist of nodes and it may be redundant to specify that the graph has nodes.